### DISCUSSION TITLE
Project Name

### DISCUSSION THREAD

**Discussion by helgeerbe** [helgeerbe/picframe#23] [Ideas]
Hi, we are coming closer to a release. Yesterday I had a discussion with Wolfgang. He asked if we could stay with the product name **PictureFrame** to point out, this is a follow up of PictureFrame2020.

He proposed that we call it PictureFrame2021, but I personally don't like the idea to have a date in the product name. I'm open to rename the project, and use a date derived release numbering. In the past I wasn't very consistent in using 'picframe' or 'picture_frame' So it might be a good point in time to do a little clean up.

- Product Name: PictureFrame
- Release#: e. g. 2021.03.01
- pypi package name: PictureFrame
- program name (executable wrapper script): picframe

What does this meen:
- Rename the repro to PictureFrame (old git references should work, but it is better so set the remote origin in git to the new URL)
- Create a new pypi package PictureFrame, and retire the old picframe
- rename picture_frame.py to picframe.py

What do you think?

**Comment by paddywwoof** [comment]
Yes I think this is the way to go. Wish I'd used a date style version number for pi3d now!

**Comment by helgeerbe** [comment]
Just checked pypi. It seems, that 'pictureframe' is already taken.

Just read the guidelines: https://www.python.org/dev/peps/pep-0423/#use-a-single-name
Stay with pypi 'picframe'?

- Product Name: PictureFrame
- Release#: e. g. 2021.03.01
- pypi package name: picframe
- modul name picframe
- program name (executable wrapper script): picframe

**Reply by sapnho** [reply]
Ah, the name discussion. A sure sign that the ship is about to sail! Perfect!

Calling it "PictureFrame" followed by the release date sounds good to me, that makes it a little bit easier to update the articles and it will be clear for everyone that this is an update. "Pi3DPictureFrame" would be a way to associate it with Pi3D and to make it indistinguishable from other "PictureFrame" apps, but it probably sounds too complicated.

@paddywwoof: Will you keep the old Pi3D code on the Pi3D demos repository if people just want to reinstall the old version?

**Reply by jgodfrey** [reply]
I'm good with whatever you guys settle on for a name, though I agree that the name and its version should be separate attributes (so, no versioning in the proper name).

Related (and mentioned elsewhere), I think it's imperative that we really get a handle on any remaining open items and prioritize them with respect to any near-term release. I'm certain there are things we'd _like_ to do that can wait. But, I'm also sure there are things that need to be done before any initial release.

We just need to get them all down so they can be considered as a whole. I may try to start some of that this evening.

**Reply by helgeerbe** [reply]
jeff, I just added some issues on my git repros. Sure, that are enhancements, that can wait. But if you have issues for the release. Please post them there, so we can track them.

**Reply by sapnho** [reply]
And yes, "picframe" as the module sounds good.

**Reply by helgeerbe** [reply]
So Paddy, your vote count. **PictureFrame**, **Pi3dPictureFrame** or **PictureFramePi3d** as project/product name?

**Reply by paddywwoof** [reply]
That's a pity. I think using PictureFrame (with capital P, F) will be problematical if there is already a pictureframe on pypi.org (see github.com/yotam/pictureframe especially as it uses PictureFrame as the main class) Also the convention is for python modules to be lower case (and file names lower case with just ClassNames upper camel style. and the non-camel versions with pi3d are very unwieldy. So, sadly, I think we have to go with picframe for the module name. But that does simplify some things as this is already being used by Helge!! I think we should try to simplify the various usage of picframe and picture_frame as much as we can, even if that means there has to be a directory picframe containing the module (directory with ```__init__.py``` inside it) picframe. Maybe the file picture_frame.py should be renamed to start.py or something like that. And convert the github repo to picframe

For 'branding' we could use the **PictureFrame** **2021.3.1** styling and headings to promote the continuity

Yes, I will leave the old PictureFrame2020 on pi3d_demos but obviously not do any enhancements.

**Reply by helgeerbe** [reply]
I'm fine with that. So I will rename the repro.

**Comment by helgeerbe** [comment]
- Renaming is done. 
- I removed all occurrences from 'picture_frame'.  
- removed the example folder from the site-package (this is home-assistant stuff. Will be moved to the wiki)
- cleaned up 'configuration_example.yaml' (set all path to `~/picframe...`
- moved mat files to `data/mat`
- cleaned model.py to get rid of obsolete mat_configs
- picture_frame.py is now start.py (`python3 -m picframe.start` should work)
- `picframe path/to/configuration.yaml` is working after picp install
- my dev branch is in sync with Paddy's develop

**Comment by helgeerbe** [comment]
Under projects you will find a kanban board for all the things that have to be done, before release.

**Comment by helgeerbe** [comment]
Guess, we are done here.
